File: python/aoc/shared.py
from pathlib import Path


def read_file(file_path: Path):
    with open(file_path, 'r') as f:
        lines = f.readlines()
        return lines


def read_file_clean(file_path: Path):
    with open(file_path, "r") as f:
        lines = [
            line.replace("\n", "") for line in f.readlines()
        ]
        return lines

File: go/aoc/shared.go
package main

import (
	"bufio"
	"log"
	"os"
)

func read_file_to_array(file_path string) ([]string, error) {
	file, err := os.Open(file_path)
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	var lines []string
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		lines = append(lines, scanner.Text())
	}

	return lines, scanner.Err()
}
